src/index.js

```diff
@@ -31,25 +29,43 @@
             const result = realm.eval(str);
             this.#realm.detach();
 
-            if (typeof result === 'function') {
-                console.log(result);
+            return this.#callableOrPrimitive(result);
+        };
 
-                return Object.freeze(result);
+        #callableOrPrimitive(value) {
+            if (typeof value === 'function') {
+                return this.#wrapCallables(value);
             }
 
-            return result;
-        };
+            if (this.#isPrimitive(value)) {
+                return value;
+            }
+
+            throw new TypeError('Cross-Realm Error: Evaluation result is not a primitive value');
+        }
+
+        #wrapCallables(connectedFn) {
+            const wrapper = this.#wrapCallables.bind(this);
+            const callableOrPrimitive = this.#callableOrPrimitive.bind(this);
+
+            return function(...args) {
+                const wrappedArgs = Array.from(args, arg => wrapper(arg));
```

Comment:
Is this the same as `args.map(wrapper)`?

Reply:
Well, how am I gonna be functional if I'm now mutating that args array?

`</joke>`